core/src/main/java/org/elasticsearch/action/ingest/PutPipelineTransportAction.java

```diff
@@ -60,7 +72,28 @@ protected WritePipelineResponse newResponse() {
 
     @Override
     protected void masterOperation(PutPipelineRequest request, ClusterState state, ActionListener<WritePipelineResponse> listener) throws Exception {
-        pipelineStore.put(clusterService, request, listener);
+        NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
```

Comment:
I think I see the value of validating pipelines for ingest pipelines etc. yet it comes with some cost, meaning every put pipeline means going to all nodes to check what processors they have. That said, put pipeline is not an api that will be called all the time, so it is probably ok and I don't see a better way to validate things at the moment.